runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapPoint.java

```diff
@@ -77,4 +79,8 @@ public void onEdgeSnap()
 	public void onEdgeUnsnap()
 	{
 	}
+
+	public void onSnapEdgeChanged(WorldMapSnapEdge newEdge)
```

Comment:
Why is this method necessary?

Reply:
To allow for the point to make changes when the snapped edge changes. For example, if I added an arrow underneath the Clue Scroll, I'd rotate the arrow in that function.

Reply:
Well, there is onEdgeSnap and unsnap. You can figure out the edge by yourself.

Reply:
onEdgeSnap and onEdgeUnsnap only fire when the icon, well, snaps and unsnaps, not when it changes what edge it's snapped onto. Without this function, there's no real way for an icon to change its look after the edge changes.

The icon snaps to the left edge, so you make an arrow that points left. Now it's snapped to the top, but it never became unsnapped. Now this icon is at the top of the screen pointing to the left.

Reply:
Just fire onEdgeSnap then instead when corner changes. Idk, imo this is not flexible at all and applies only at once specific scenario, so i dont really see a reason why it should be global thing. You can do that also from inside of ClueScrollPlugin.

Reply:
Because it will be useful in 1 specific scenario. And you can do what you want to do here with existing API (there is method for transforming world point to map relative point, that is public and you can easily use for determining if you are at corner or not).